js/main.js

```diff
@@ -14,26 +14,49 @@ var $entryList = document.querySelector('ul');
 
 $form.addEventListener('submit', function (event) {
   event.preventDefault();
-  var newEntry = {
-    entryId: data.nextEntryId,
-    note: $form.elements.notes.value,
-    link: $form.elements.photourl.value,
-    title: $form.elements.title.value
-  };
-  data.nextEntryId++;
-  data.entries.unshift(newEntry);
-  document.querySelector('#default-pic').src = 'images/placeholder-image-square.jpg';
-  $form.reset();
-  toggleNoEntries();
-  viewSwap('entries');
-  var newLiReturn = renderEntry(newEntry);
-  $entryList.appendChild(newLiReturn);
+
+  if (data.editing === null) {
+    var newEntry = {
+      entryId: data.nextEntryId,
+      note: $form.elements.notes.value,
+      link: $form.elements.photourl.value,
+      title: $form.elements.title.value
+    };
+    data.entries.unshift(newEntry);
+    document.querySelector('#default-pic').src = 'images/placeholder-image-square.jpg';
+    $form.reset();
+    toggleNoEntries();
+    viewSwap('entries');
+    var newLiReturn = renderEntry(newEntry);
+    $entryList.appendChild(newLiReturn);
+    data.nextEntryId++;
+  } else {
+    var editingIndex = data.entries.entryId;
+    var editedEntry = {
+      entryId: editingIndex,
+      note: $form.elements.notes.value,
+      link: $form.elements.photourl.value,
+      title: $form.elements.title.value
+    };
+
+    var $allTheLi = document.querySelectorAll('li');
+    for (var i = 0; i < $allTheLi.length; i++) {
+      var individualLi = Number($allTheLi[i].getAttribute('data-entry-id'));
+      if (data.editing.entryId === individualLi) {
+        $allTheLi[i].replaceWith(renderEntry(editedEntry));
+      }
+    }
+    toggleNoEntries();
+    viewSwap('entries');
+    data.editing = null;
+  }
```

Comment:
It doesn't look like you are updating the edited entry's data in `data.entries`. Does your edit persist after you refresh the page?